cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java

```diff
@@ -168,6 +170,31 @@ public void query(HttpRequest request, HttpResponder responder,
     }
   }
 
+  @POST
+  @Path("/delete")
+  public void delete(HttpRequest request, HttpResponder responder,
+                    @QueryParam("context") String context,
+                    @QueryParam("metric") String metric,
+                    @QueryParam("start") @DefaultValue("0") final long startTs,
+                    @QueryParam("end") @DefaultValue("2147483647") final long endTs) throws Exception {
+    try {
+      if (context == null) {
+        responder.sendString(HttpResponseStatus.BAD_REQUEST, "Please Specify context to delete");
+        return;
+      }
+      Map<String, String> tagsSliceBy = humanToTagNames(parseTagValuesAsMap(context));
+      MetricDeleteQuery query = new MetricDeleteQuery(startTs, endTs, metric, tagsSliceBy);
+      metricStore.delete(query);
+      responder.sendStatus(HttpResponseStatus.OK);
+    } catch (IllegalArgumentException e) {
```

Comment:
Where does this get thrown? Better to respond with ``BAD_REQUEST`` only for known ``IllegalArgumentException``s and not all?